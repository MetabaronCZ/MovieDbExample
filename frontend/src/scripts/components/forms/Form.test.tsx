import { act } from '@testing-library/react';
import { fireEvent, getByText } from '@testing-library/dom';
import { describe, expect, it, jest } from '@jest/globals';

import { Form } from './Form';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/forms/Form', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <Form onSubmit={() => null}>CONTENT</Form>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const form = getByText(container, 'CONTENT');
    expect(form).toBeInTheDocument();
    expect(form.tagName.toLowerCase()).toEqual('form');
  });

  it('should render loading state', async () => {
    const { container } = await waitForComponent(<Form loading>CONTENT</Form>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const form = getByText(container, 'CONTENT');
    expect(form).toBeInTheDocument();

    const wrapper = form.parentNode;
    expect(wrapper).toBeInTheDocument();
    expect(wrapper).toHaveStyle({ opacity: 0.65 });
  });

  it('should callback submit event', async () => {
    const onSubmit = jest.fn();

    const { container } = await waitForComponent(
      <Form onSubmit={onSubmit}>
        <button type="submit">SUBMIT</button>
      </Form>,
      { wrapper: TestComponentWrapper },
    );
    expect(onSubmit).not.toHaveBeenCalled();

    const button = getByText(container, 'SUBMIT');
    expect(button).toBeInTheDocument();

    // click submit button
    act(() => {
      fireEvent.click(button);
    });

    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
