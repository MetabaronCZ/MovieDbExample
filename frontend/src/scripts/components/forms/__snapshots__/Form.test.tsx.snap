// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/Form should render loading state 1`] = `
.c0 {
  opacity: 0.65;
  pointer-events: none;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

<div>
  <div
    class="c0"
  >
    <form
      class="c1"
    >
      CONTENT
    </form>
  </div>
</div>
`;

exports[`components/forms/Form should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

<div>
  <form
    class="c0"
  >
    CONTENT
  </form>
</div>
`;
